aos_cd_jobs/updater.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python
+from os import getenv, listdir, rename, walk
+from os.path import join, relpath
+from shutil import rmtree
+
+from git import Repo
+
+from aos_cd_jobs.common import JOBS_DIRECTORY, initialize_repo
+
+def update_branches(repo):
+    for job in list_jobs(repo):
+        if job in repo.branches:
+            branch = repo.branches[job]
+        else:
+            branch = None
+        if branch is not None:
+            branch.delete()
+        create_remote_branch(repo, job)
+
+def list_jobs(repo):
+    jobs = []
+    jobs_directory = join(repo.working_dir, JOBS_DIRECTORY)
+    for (dirpath, _, filenames) in walk(jobs_directory):
+        if 'Jenkinsfile' in filenames:
+            jobs.append(relpath(dirpath, jobs_directory))
+    return jobs
+
+def create_remote_branch(repo, name):
+    branch = repo.heads.master.checkout(orphan=name)
+    directory = join(repo.working_dir, JOBS_DIRECTORY, name)
+    create_job_file_tree(repo.working_dir, directory)
+    repo.index.remove(repo.index.entries.values())
+    repo.index.add([f for f in listdir(repo.working_dir) if f != '.git'])
```

Comment:
I think the suggestions in [this thread](https://github.com/gitpython-developers/GitPython/issues/292) are a little clearer and more correct than this.